src/main/java/com/strandls/document/service/Impl/DocumentServiceImpl.java

```diff
@@ -1158,4 +1207,155 @@ public Boolean documentDownloadLog(HttpServletRequest request, DownloadLogData d
 		return false;
 	}
 
+	@Override
+	public void produceToRabbitMQ(String documentData, String documentId) {
+		try {
+			producer.setMessage("esmodule", documentData, documentId);
+		} catch (Exception e) {
+			logger.error(e.getMessage());
+		}
+
+	}
+
+	@Override
+	public DocumentListData getDocumentList(String index, String type, String geoAggregationField,
```

Comment:
move this list page generation code to a seperate interface and impl

Reply:
createb new impl /interface for list and bind it